File: AndreasH-julia/Day3/part1.jl
using Printf

function readfile(filename)
    try
        open(filename,"r") do io
            inputs = readlines(io)
            return inputs
        end;
    catch
        print("Error parsing file")
        return []
    end
end

function takeStep(map,position)
    try
        forrestLine= map[position[2]]
        xPos = mod(position[1],length(map[1])) == 0 ? length(map[1]) : mod(position[1],length(map[1]))
        value = forrestLine[xPos]
        if string(value) == "#"
            global treeCount += 1
        end
        takeStep(map,position + stepSize)
    catch 
        return
    end
end


forrest = readfile("input.txt")
stepSize = [3,1]
treeCount = 0
takeStep(forrest,[1,1])
@printf("\nNumber of trees encountered: %d",treeCount)

File: nattfalk-python/day1/part1.py
list_file = open("input.txt", "r")
lines = list_file.readlines()
list_file.close()

for x in lines:
    for y in reversed(lines):
        if int(x) + int(y) == 2020:
            break
    else:
        continue
    break

print(x.rstrip() + "*" + y.rstrip() + "=" + str(int(x)*int(y)))
